src/librustc_resolve/resolve_imports.rs

```diff
@@ -627,14 +627,16 @@ impl<'a, 'b> ImportResolver<'a, 'b> {
             self.finalize_resolutions_in(module);
         }
 
-        let mut has_errors = false;
         let mut seen_spans = FxHashSet::default();
         let mut errors = vec![];
         let mut prev_root_id: NodeId = NodeId::from_u32(0);
-        for i in 0 .. self.r.determined_imports.len() {
-            let import = self.r.determined_imports[i];
+        let mut determined_imports = vec![];
+        mem::swap(&mut self.r.determined_imports, &mut determined_imports);
```

Comment:
```suggestion
        let determined_imports = mem::take(&mut self.r.determined_imports);
```
(and `indeterminate_imports` too)